vscode-dotty/src/worksheet.ts

```diff
@@ -44,14 +44,15 @@ class Worksheet {
    * removes redundant blank lines that have been inserted by a previous
    * run.
    */
-  prepareForRunning(): void {
+  prepareRun(): void {
     this.removeRedundantBlankLines().then(_ => this.reset())
   }
```

Comment:
I think that `prepareRun` should return a `Thenable<{}>`, and we should wait for it to complete before we continue in `run`. We should also return that `Thenable` in `willSaveTextDocument` so that VSCode waits for it to complete before saving.